kasa/tapo/childdevice.py

```diff
@@ -24,21 +25,27 @@ def __init__(
         self._parent = parent
         self._id = child_id
         self.protocol = _ChildProtocolWrapper(child_id, parent.protocol)
+        # TODO: remove the assignment after modularization is done,
+        #  currently required to allow accessing time-related properties
+        self._time = parent._time
+        # TODO: cleanup the _last_update, _sys_info, _info, _data mess.
+        self._last_update = (
+            self._sys_info
+        ) = self._info = self._data = self._get_child_info()
+        self._device_type = DeviceType.Plug
+
+    def _get_child_info(self) -> Dict:
+        """Return the device specific information from the parent."""
+        for child in self._parent._last_update["child_info"]["child_device_list"]:
+            if child["device_id"] == self._id:
+                return child
+
```

Comment:
Its a shame that we have to linear search for the device_id for every child but it likely doesn't matter since the child list is never likely to be longer than 6

Reply:
Indeed, I think I'm going to restructure the child handling right away to get the data pushed as a part of parent's update cycle. There should be no need for having but a noop `update()` in childdevice.

Reply:
The internal state is now updated by the parent. #733 will fix the tests.